src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,54 @@
+package duke.task;
+
+/**
+ * Represents a task.
+ */
+public abstract class Task {
+    private String name;
+    private boolean isDone;
+
+    protected Task(String name) {
+        this(name, false);
+    }
+
+    protected Task(String name, boolean isDone) {
+        this.name = name;
+        this.isDone = isDone;
+    }
+
+    protected Task markAsDone() {
+        isDone = true;
+        return this;
+    }
+
+    /**
+     * Returns the name of the task.
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Returns whether the task is marked as completed.
+     */
+    public boolean checkTaskDone() {
```

Comment:
```suggestion
    public boolean isDone() {
```

Reply:
I agree with this change as well

Reply:
Thanks for the suggestion. Have changed it! 😄